### PR TITLE
[keyboard] Adding new keyboard UD_40 ortho version

### PR DESCRIPTION
**Pull request by dejaeghered**
## Description

Added PCB for the Ortho version for keyboard UD_40 (directory: mechstudio/ud_40_ortho)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechstudio/ud_40_ortho/info.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "keyboard_name": "UD_40 ortho", 
+  "url": "https://mech.studio/projects/ud_40/", 
+  "maintainer": "dejaeghered", 
+  "width": 12, 
+  "height": 4, 
+  "layouts": {
+    "LAYOUT": {
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/info.json:8]:
```suggestion
    "LAYOUT_ortho_4x12": {
```

File: keyboards/mechstudio/ud_40_ortho/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+#define _MA 0 //Main layer
+#define _LO 1 //Low
+#define _HI 2 //High
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[_MA] = LAYOUT(
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/keymaps/default/keymap.c:28]:
```suggestion
[_MA] = LAYOUT_ortho_4x12(
```

File: keyboards/mechstudio/ud_40_ortho/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  KC_ESC,   KC_Q,        KC_W,        KC_E,        KC_R,        KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_BSPC,
+  KC_TAB,   KC_A,        KC_S,        KC_D,        KC_F,        KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,
+  KC_LSFT,  KC_Z,        KC_X,        KC_C,        KC_V,        KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  KC_ENT,
+  _______,  KC_LCTRL,    KC_LALT,     KC_LGUI,     MO(_LO),     KC_SPC,   KC_SPC,   MO(_HI),  KC_LEFT,  KC_DOWN,  KC_UP,    KC_RGHT),
+
+[_LO] = LAYOUT(
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/keymaps/default/keymap.c:34]:
```suggestion
[_LO] = LAYOUT_ortho_4x12(
```

File: keyboards/mechstudio/ud_40_ortho/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  KC_GRV,   KC_1,        KC_2,        KC_3,        KC_4,        KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_DEL,
+  KC_TILD,  KC_F1,       KC_F2,       KC_F3,       KC_F4,       KC_F5,    KC_F6,    KC_MINS,  KC_EQL,   KC_LBRC,  KC_RBRC,  KC_BSLS,
+  _______,  KC_F7,       KC_F8,       KC_F9,       KC_F10,      KC_F11,   KC_F12,   _______,  _______,  KC_PGUP,  KC_PGDN,  _______,
+  KC_INS,   KC_HOME,     KC_END,      KC_DEL,      _______,     _______,  _______,  _______,  _______,  KC_VOLD,  KC_VOLU,  _______),
+
+[_HI] = LAYOUT(
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/keymaps/default/keymap.c:40]:
```suggestion
[_HI] = LAYOUT_ortho_4x12(
```

File: keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT(
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c:19]:
```suggestion
[0] = LAYOUT_ortho_4x12(
```

File: keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  KC_ESC,   KC_Q,        KC_W,        KC_E,        KC_R,        KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_BSPC,
+  KC_TAB,   KC_A,        KC_S,        KC_D,        KC_F,        KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,
+  KC_LSFT,  KC_Z,        KC_X,        KC_C,        KC_V,        KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  KC_ENT,
+  _______,  KC_LCTRL,    KC_LALT,     KC_LGUI,     MO(1),       KC_SPC,   KC_SPC,   MO(2),    KC_LEFT,  KC_DOWN,  KC_UP,    KC_RGHT),
+
+[1] = LAYOUT(
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c:25]:
```suggestion
[1] = LAYOUT_ortho_4x12(
```

File: keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT(
+  KC_ESC,   KC_Q,        KC_W,        KC_E,        KC_R,        KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_BSPC,
+  KC_TAB,   KC_A,        KC_S,        KC_D,        KC_F,        KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,
+  KC_LSFT,  KC_Z,        KC_X,        KC_C,        KC_V,        KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  KC_ENT,
+  _______,  KC_LCTRL,    KC_LALT,     KC_LGUI,     MO(1),       KC_SPC,   KC_SPC,   MO(2),    KC_LEFT,  KC_DOWN,  KC_UP,    KC_RGHT),
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c:23]: Via supports tri layer with specific keycodes. 
```suggestion
  _______,  KC_LCTRL,    KC_LALT,     KC_LGUI,     KC_FN13,     KC_SPC,   KC_SPC,   KC_FN23,  KC_LEFT,  KC_DOWN,  KC_UP,    KC_RGHT),
```

File: keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+[1] = LAYOUT(
+  KC_GRV,   KC_1,        KC_2,        KC_3,        KC_4,        KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_DEL,
+  KC_TILD,  KC_F1,       KC_F2,       KC_F3,       KC_F4,       KC_F5,    KC_F6,    KC_MINS,  KC_EQL,   KC_LBRC,  KC_RBRC,  KC_BSLS,
+  _______,  KC_F7,       KC_F8,       KC_F9,       KC_F10,      KC_F11,   KC_F12,   _______,  _______,  KC_PGUP,  KC_PGDN,  _______,
+  KC_INS,   KC_HOME,     KC_END,      KC_DEL,      _______,     _______,  _______,  _______,  _______,  KC_VOLD,  KC_VOLU,  _______),
+
+[2] = LAYOUT(
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c:31]:
```suggestion
[2] = LAYOUT_ortho_4x12(
```

File: keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+  _______,  RGB_TOG,  RGB_MOD,  RGB_HUI,  RGB_SAI,  RGB_VAI,  _______,  _______,  _______,  _______,  _______,  _______,
+  _______,  _______,  RGB_RMOD, RGB_HUD,  RGB_SAD,  RGB_VAD,  _______,  _______,  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______),
+  
+[3] = LAYOUT(
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/keymaps/via/keymap.c:37]:
```suggestion
[3] = LAYOUT_ortho_4x12(
```

File: keyboards/mechstudio/ud_40_ortho/ud_40_ortho.h [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/ud_40_ortho.h:30]:
```suggestion
#define LAYOUT_ortho_4x12( \
```

File: keyboards/mechstudio/ud_40_ortho/ud_40_ortho.h [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+) \
+{ \
+	{ k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b }, \
+	{ k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b }, \
+	{ k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b }, \
+	{ k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b } \
+}
+
+void matrix_init_user(void);
+void matrix_scan_user(void);
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/ud_40_ortho.h:44]: These shouldn't be needed. 
```suggestion
```

File: keyboards/mechstudio/ud_40_ortho/rules.mk [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+ENCODER_ENABLE = no
```
Comment (drashna) [inline keyboards/mechstudio/ud_40_ortho/rules.mk:31]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output 
ENCODER_ENABLE = no

LAYOUTS = ortho_4x12
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by dejaeghered** [comment]
@drashna thank you for reviewing.  I merged the suggestions and resolved some syntax issues.

File: keyboards/mechstudio/ud_40_ortho/config.h [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4D53
+#define PRODUCT_ID      0x0002
+#define DEVICE_VER      0x0002
+#define MANUFACTURER    "Mech Studio"
+#define PRODUCT         "UD_40 Ortholinear"
+#define DESCRIPTION     "A 40% keyboard by Goldworms"
```
Comment (zvecr) [inline keyboards/mechstudio/ud_40_ortho/config.h:27]:
```suggestion
```

File: keyboards/mechstudio/ud_40_ortho/rules.mk [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+MCU = atmega32u2
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (zvecr) [inline keyboards/mechstudio/ud_40_ortho/rules.mk:11]:
```suggestion
```

File: keyboards/mechstudio/ud_40_ortho/ud_40_ortho.h [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+#define XXX KC_NO
```
Comment (zvecr) [inline keyboards/mechstudio/ud_40_ortho/ud_40_ortho.h:19]: Unused
```suggestion
```

File: keyboards/mechstudio/ud_40_ortho/rules.mk [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output 
+ENCODER_ENABLE = no
```
Comment (zvecr) [inline keyboards/mechstudio/ud_40_ortho/rules.mk:31]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/mechstudio/ud_40_ortho/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+#define _MA 0 //Main layer
+#define _LO 1 //Low
+#define _HI 2 //High
```
Comment (zvecr) [inline keyboards/mechstudio/ud_40_ortho/keymaps/default/keymap.c:25]:
```suggestion
// Defines names for use in layer keycodes and the keymap
enum layer_names {
    _MA,
    _LO,
    _HI
};
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/mechstudio/ud_40_ortho/config.h [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4D53
+#define PRODUCT_ID      0x0002
+#define DEVICE_VER      0x0002
+#define MANUFACTURER    "Mech Studio"
+#define PRODUCT         "UD_40 Ortholinear"
```
Comment (fauxpark) [inline keyboards/mechstudio/ud_40_ortho/config.h:26]:
```suggestion
#define MANUFACTURER    Mech Studio
#define PRODUCT         UD_40 Ortholinear
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**